OFFICAL BAN OVERHAUL PR

Halcyon- opened the pull request:
+Fixed stray apostrophe left in ban redirect.
+Deleted .htaccess based banning, switched to SQL based banning.
+Fixed a bug where deleting parent posts did not remove the children from the management panel or database
+Check if the user is banned everytime they post
+Reordered SQL config values info for no reason
+Every ban is perma-posting ban for now, unless someone goes into the
SQL table and manually deletes the entry

COMING SOON:
-Ability to remove bans
-Ban redirect page with reason why, IP banned from
-Automated ban removal after time served is up
- code is hacky as SHIT
  Hopefully I'll have something else neat done tonight ¯_(ツ)_/¯

Apogate committed:
Fixed ban form (again) + jquery things

+Missed a apostrophe  in the ban form
+Default config setup accidentally disabled RePod's beautiful JQuery
suite & messed up the exit icons, fixed relevant filepaths

Commit:
Update suite_settings.js

Commit:
Update suite_settings.js

Commit:
Update suite_settings.js

RePod commented on config.php:
```diff
@@ -81,7 +81,7 @@
 define(PHP_SELF2, 'index.html');						//Name of main htm file
 define(PHP_EXT, '.html');								//Extension used for board pages after first
-define(JS_PATH, 'jquery'); 								//relative path from imgboard.php of the jquery folder without a trailing slash
+define(JS_PATH, 'plugins/jquery'); 								//relative path from imgboard.php of the jquery folder without a trailing slash
```
Wouldn't it be better to define this after PLUG_PATH then inherit it?

Halcyon- replied:
Whatever's easiest and does the least damage, I didn't want to mess with the suite

Apogate committed:
Extreme Saguaro makeover: ding dong bannu edition

+Bans are now SQL based, another required define in config for
SQLBANLOG, holds ban information
+Check if the user is banned everytime they post
+Reordered SQL config values info for no reason
+Every ban is perma-posting ban for now, unless someone goes into the
SQL table and manually deletes the entry

COMING SOON:
-Ability to remove bans
-Ban redirect page with reason why, IP banned from
-Automated ban removal after time served is up
- code is hacky as SHIT

-------RINE ENDS HERE

Apogate committed:
part 2: the editing

whoops

Apogate committed:
part 3: return of the editing

Always proofread your work, kids

Apogate committed:
Fixed a deletion bug

Fixed a bug where deleting an OP post did not remove the children from
the management table or SQL database

Halcyon- commented:
I'll leave all this here to be merged in case I get hit by a bus or something and disappear. Otherwise, don't merge this yet until the ban system is complete. I'll keep adding to it as I go along.

Apogate committed:
part 3: son of the editing

fixed a bug where i'm retarded and ruin my own code

Apogate committed:
Makes reply thumbnails smaller than OP thumbnails

Minor fixes to bans...again....

Apogate committed:
Foundation for ban auto expiring bans

Apogate committed:
Merge this sick filth

Ready for merge, recap on new features:
***New
+Ability to warn users for posts/rule violations

+SQL ban based system
+Checks if user is banned before each post is submitted
+Redirect  users to banned.php, displays reason and time
+Snuck PROPER THUMBNAIL SCALING for replies in there, holy hell that
bothered me as soon as I realized they all scaled to the same standard
250x250 pixels

-Bans auto deleting wont be implemented until the management panel
overhaul, which is in the works

This PR is done
<3 Halcyon

Apogate committed:
-

Halcyon- commented:
I also recommend deleting kusaba and monotone .css and adding them to the CSS fun pack dlc, they're kinda old and I don't know how popular they'd be.

RePod commented:
Not really translation friendly at the moment, but I guess that's the biggest problem with having something support translations.

Apogate committed:
--
